core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/load/withLoad.ts

```diff
@@ -32,68 +32,61 @@ export function withLoad() {
         },
         withClient(),
         withMethods((store) => {
+            const router = inject(Router);
             const dotPageApiService = inject(DotPageApiService);
             const dotLanguagesService = inject(DotLanguagesService);
             const dotLicenseService = inject(DotLicenseService);
-            const loginService = inject(LoginService);
             const dotExperimentsService = inject(DotExperimentsService);
-            const router = inject(Router);
-            const activatedRoute = inject(ActivatedRoute);
+            const loginService = inject(LoginService);
 
             return {
-                init: rxMethod<DotPageApiParams>(
+                /**
+                 * Fetches the page asset based on the provided page parameters.
+                 * This method updates the user permissions, pageAsset, and pageParams.
+                 *
+                 * @param {DotPageApiParams} pageParams - The parameters used to fetch the page asset.
+                 * @memberof DotEmaShellComponent
+                 */
+                loadPageAsset: rxMethod<Partial<DotPageApiParams>>(
                     pipe(
                         tap(() => store.resetClientConfiguration()),
                         tap(() => {
                             patchState(store, { status: UVE_STATUS.LOADING, isClientReady: false });
                         }),
                         switchMap((params) => {
+                            const pageParams = {
+                                ...(store.pageParams() ?? {}),
+                                ...params
+                            } as DotPageApiParams;
```

Comment:
```suggestion
                            if (!params.url && !store.pageParams()?.url) {
                                throw new Error('URL is required');
                            }

                            const pageParams: DotPageApiParams = {
                                url: '', // default value
                                language_id: store.pageParams().language_id, // default value
                                'com.dotmarketing.persona.id': 'default', // default value if required
                                ...(store.pageParams() ?? {}),
                                ...params
                            };
```

`as` is fake 😂

Reply:
Yeah, that was part of what I wanted to discuss last Friday. It was always fake ahh

Reply:
That was part of what I wanted to discuss last Friday.

Reply:
Yeah we need to stop the `as`